packages/medical-domain/src/domains/models/medical/datum/basics/base-datum.model.ts

```diff
@@ -25,14 +25,38 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import type Source from '../enums/source.enum'
+import Source from '../enums/source.enum'
 import type BaseTime from './base-time.model'
-import { type DatumType } from '../enums/datum-type.enum'
+import { isBaseTime } from './base-time.model'
+import { DatumType } from '../enums/datum-type.enum'
+import { isRecord } from '../../../../utils/typeguard.utils'
 
 type BaseDatum = BaseTime & {
   id: string
   type: DatumType
   source: Source
 }
 
+function isBaseDatum(value: unknown): value is BaseDatum {
+  if (isBaseTime(value) && isRecord(value)) {
```

Comment:
This type guard automatically casts value to the type `BaseTime & Record<string, unknown>`.If i only check `isBaseTime(value)` it will be casted to `BaseTime` only.

So in order to test  other fields like id, type & source. I would have to do an additional cast i.e. redeclare `const recordValue = value as Record<string, unknown> ` to do all the tests after.